be/src/vec/functions/array/function_array_remove.h

```diff
@@ -322,13 +330,21 @@ class FunctionArrayRemove : public IFunction {
                 res = _execute_number_expanded<ColumnDecimal128>(offsets, *nested_column,
                                                                  *right_column, nested_null_map);
             }
-        } else if (is_date_or_datetime(right_type) && is_date_or_datetime(left_element_type)) {
+        } else if ((is_date_or_datetime(right_type) && is_date_or_datetime(left_element_type)) ||
+                   (is_date_v2_or_datetime_v2(right_type) &&
+                    is_date_v2_or_datetime_v2(left_element_type))) {
```

Comment:
Why not use `_execute_number_expanded<ColumnDateV2>` if `is_date_v2_or_datetime_v2(right_type)` ?